app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -540,222 +557,218 @@ fun AgendaScreenContent(
+            }
+        }
+
         // • SHOW AGENDA ITEMS LIST
-        if (agendaItems.isNotEmpty()) {
-            LazyColumn(
-                state = scrollState,
-                modifier = Modifier
-                    .background(color = MaterialTheme.colors.surface)
-                    .fillMaxSize()
-                    .padding(start = DP.tiny, end = DP.tiny)
+        LazyColumn(
+            state = scrollState,
+            modifier = Modifier
+                .background(color = MaterialTheme.colors.surface)
+                .fillMaxSize()
+                .padding(start = DP.tiny, end = DP.tiny)
 
-            ) {
-                val sortedAgendaItems = agendaItems.sortedBy { agendaItem ->
-                    agendaItem.startTime
-                }
-                val agendaItemsBeforeNow = sortedAgendaItems.filter { agendaItem ->
-                    agendaItem.startTime.isBefore(zonedDateTimeNow)
-                }
-                val agendaItemsAfterNow = sortedAgendaItems.filter { agendaItem ->
-                    agendaItem.startTime.isAfter(zonedDateTimeNow)
-                }
+        ) {
+            val sortedAgendaItems = agendaItems.sortedBy { agendaItem ->
+                agendaItem.startTime
+            }
+            val agendaItemsBeforeNow = sortedAgendaItems.filter { agendaItem ->
+                agendaItem.startTime.isBefore(zonedDateTimeNow)
+            }
+            val agendaItemsAfterNow = sortedAgendaItems.filter { agendaItem ->
+                agendaItem.startTime.isAfter(zonedDateTimeNow)
+            }
 
-                fun isToday() = (
-                        weekStartDate.plusDays(selectedDayIndex?.toLong() ?: 0).year ==
-                                ZonedDateTime.now().year
-                                && weekStartDate.plusDays(
-                            selectedDayIndex?.toLong() ?: 0
-                        ).dayOfYear ==
-                                ZonedDateTime.now().dayOfYear
-                        )
+            fun isToday(): Boolean {
+                val selectedDay = selectedDayIndex?.toLong() ?: 0
+                val yearOfSelectedDay = weekStartDate.plusDays(selectedDay).year
+                val dayNumberOfSelectedDay = weekStartDate.plusDays(selectedDay).dayOfYear
 
-                if (!isToday() || (isToday() && agendaItemsBeforeNow.isNotEmpty())) {
-                    item("before_spacer_for_today") {
-                        Spacer(
-                            modifier = Modifier
-                                .height(14.dp)
-                                .fillMaxWidth()
-                        )
-                    }
+                return (
+                    yearOfSelectedDay == ZonedDateTime.now().year
+                    &&
+                    dayNumberOfSelectedDay == ZonedDateTime.now().dayOfYear
+                )
+
+            }
+
+            if (!isToday() || (isToday() && agendaItemsBeforeNow.isNotEmpty())) {
```

Comment:
Export this on a variable or method. It makes it easier to understand

Reply:
ok will improve and refactor